sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1279,4 +1279,62 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
       Row("0,2013-111-11 12:13:14") :: Row(null) :: Nil
     )
   }
+
+  def checkHeader(multiLine: String): Unit = {
+    test(s"SPARK-23786: Checking column names against schema ($multiLine)") {
```

Comment:
Below is preferred, since it is easier for us to run the test in IntelliJ

```
test("SPARK-23786: Checking column names against schema (multiline )") {
  checkHeader(multiLine = true)
}
```